app/src/org/commcare/android/database/connect/models/ConnectJobRecord.java

```diff
@@ -403,10 +404,10 @@ public Hashtable<String, Integer> getDeliveryCountsPerPaymentUnit(boolean todayO
         for(int i = 0; i < deliveries.size(); i++) {
             ConnectJobDeliveryRecord delivery = deliveries.get(i);
             int oldCount = 0;
-            if(paymentCounts.containsKey(delivery.getSlug())) {
-                oldCount = paymentCounts.get(delivery.getSlug());
-            }
-
+            boolean isSameDay=sameDay(new Date(),delivery.getDate());
+                if(paymentCounts.containsKey(delivery.getSlug()) && (isSameDay && todayOnly)) {
```

Comment:
I think this:
(isSameDay && todayOnly)
needs to change to:
(isSameDay || !todayOnly)